### DISCUSSION TITLE
can a client also be a server?

### DISCUSSION THREAD

**Discussion by pavlo-yashchuk** [FreeOpcUa/python-opcua#1453] [Q&A] [answered]
Hi all
How do you think, can a client also be a server?
have API which starts on threads a few clients connection
Can I add on that API an opc ua server?

**Comment by AndreasHeine** [comment, marked as answer]
typically not this way!

https://reference.opcfoundation.org/Core/docs/Part1/6.1/
https://reference.opcfoundation.org/Core/docs/Part1/6.3.7/
as spec says a OPC UA Server Application can act like a OPC UA Client via an Client-Interface (Client-Class-Instance)

**Reply by pavlo-yashchuk** [reply]
@AndreasHeine thanks!
